inference-engine/src/vpu/graph_transformer/src/stages/strided_slice.cpp

```diff
@@ -35,36 +52,95 @@ class StridedSliceStage final : public StageNode {
     }
 
     void initialCheckImpl() const override {
-        IE_ASSERT(numInputs() == 3 || numInputs() == 4);
-        IE_ASSERT(numOutputs() == 1);
+        VPU_THROW_UNLESS(numInputs() == 3 || numInputs() == 4,
+            "Validating layer {} with type {} failed: number of input should be 3 or 4, but {} were provided",
+            name(), type(), numInputs());
+        VPU_THROW_UNLESS(numOutputs() == 1,
+            "Validating layer {} with type {} failed: number of outputs should be 1, but {} were provided",
+            name(), type(), numOutputs());
+
+        std::string newAxisMask = origLayer()->GetParamAsString("new_axis_mask", "");
+        VPU_THROW_UNLESS(maskStrToInt(newAxisMask) == 0,
+            "Checking {} with type {} failed: new_axis_mask parameter is not supported",
+            name(), type());
+        std::string shrinkAxisMask = origLayer()->GetParamAsString("shrink_axis_mask", "");
+        VPU_THROW_UNLESS(maskStrToInt(newAxisMask) == 0,
+            "Checking {} with type {} failed: shrink_axis_mask parameter is not supported",
+            name(), type());
+        std::string ellipsisMask = origLayer()->GetParamAsString("ellipsis_mask", "");
+        VPU_THROW_UNLESS(maskStrToInt(newAxisMask) == 0,
+            "Checking {} with type {} failed: ellipsis_mask parameter is not supported",
+            name(), type());
+
+        const auto& strides = input(3);
+        if (strides->usage() == DataUsage::Const) {
+            const auto stridesPtr = strides->content()->get<int32_t>();
+            VPU_THROW_UNLESS(stridesPtr != nullptr,
+                             "Checking {} with type {} failed: pointer for strides is null");
+            for (int i = 0; i < strides->desc().totalDimSize(); i++) {
+                VPU_THROW_UNLESS(stridesPtr[i] > 0,
+                                 "Checking {} with type {} failed: negative stride is not supported");
+            }
+        }
 
         std::vector<EnumSet<DataType>> expectedInputs3Types =
-            { {DataType::FP16}, {DataType::S32}, {DataType::S32} };
+            { {DataType::FP16, DataType::S32}, {DataType::S32}, {DataType::S32} };
```

Comment:
Does StridedSlice support only FP16 and S32? Is it related on data type?

Reply:
It depends on expected inputs of `Copy` but I see it can process everything. Fixed.